compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -630,7 +630,7 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
             }
         }
         compareTypeLambda
-      case tp2 as OrType(tp21, tp22) =>
+      case tp2 @ OrType(tp21, tp22) =>
```

Comment:
Are we also reverting the possibility to use `as` instead of `@` ?

Reply:
no there are other pattern match bindings later on that still use `as`

Reply:
Thanks! I was super confused by this change.